verifier/src/main/java/com/redhat/ipaas/verifier/ForkingVerifier.java

```diff
@@ -0,0 +1,177 @@
+        if( response != null ) {
+            LinkedHashMap<String, Error> errors = new LinkedHashMap<>();
+            for (Map.Entry<Object, Object> entry : response.entrySet()) {
+                String key = (String) entry.getKey();
+                if( key.startsWith("error.") ) {
+                    String errorId = key.substring("error.".length()).replaceFirst("\\..*", "");
+                    Error error = errors.getOrDefault(errorId, createError());
+                    String value = (String) entry.getValue();
+                    if( key.endsWith(".code") ) {
+                        error = error.withCode(value);
+                    }
+                    if( key.endsWith(".description") ) {
+                        error = error.withDescription(value);
+                    }
+                    errors.put(errorId, error);
+                }
+            }
+            builder.addAllErrors(errors.values());
+        }
+        return builder.build();
+    }
+
+    private Error createError() {
+        return ImmutableError.builder().build();
+    }
+
+    private Properties runValidator(String classpath, Scope scope, String camelPrefix, Properties request) throws IOException, InterruptedException {
+        Process java = new ProcessBuilder()
+            .command(
+                "java", "-classpath", classpath,
+                "com.redhat.ipaas.connector.ConnectorVerifier", scope.toString(), camelPrefix
+            )
+            .redirectError(ProcessBuilder.Redirect.INHERIT)
+            .start();
+
+        try (OutputStream os = java.getOutputStream()) {
+            request.store(os, null);
+        }
+        Properties result = new Properties();
+        try (InputStream is = java.getInputStream()) {
+            result.load(is);
+        }
+
+        if (java.waitFor() != 0) {
+            throw new IOException("Verifier failed with exit code: " + java.exitValue());
+        }
+        return result;
+    }
+
+
+    private String getConnectorClasspath(Connector connector) throws IOException, InterruptedException {
+        byte[] pom = projectGenerator.generatePom(connector);
+        java.nio.file.Path tmpDir = Files.createTempDirectory("ipaas-connector");
+        try {
+            Files.write(tmpDir.resolve("pom.xml"), pom);
+            ArrayList<String> args = new ArrayList<>();
+            args.add("mvn");
+            args.add("org.apache.maven.plugins:maven-dependency-plugin:3.0.0:build-classpath");
+            if (localMavenRepoLocation != null) {
+                args.add("-Dmaven.repo.local=" + localMavenRepoLocation);
+            }
+            ProcessBuilder builder = new ProcessBuilder().command(args)
+                    .redirectError(ProcessBuilder.Redirect.INHERIT)
+                    .directory(tmpDir.toFile());
+            Map<String, String> environment = builder.environment();
+            environment.put("MAVEN_OPTS", "-Xmx64M");
+            Process mvn = builder.start();
```

Comment:
What are plans for the future ? Starting an extra JVM for each verification request will blow up the ipaas-rest pod quite soon. 10 parallel requests == 10 JVM started simultaneously in a memory restricted pod ?

tbh, I'm really a fan of 1 container = 1 process (although systemd ninja will disagree ;-). But starting externally processes boundlessly in the main rest-server is a call for disaster.

I know its for getting quick results for this sprint but I hope we all agree that expensive operations likes this should be run in one or more different resource-controlled pods.